[processing / tilesxyzdirectory] Add some options for Leaflet html output

pathmapper opened the pull request:
Three new optional advanced parameters:

- Title
- Attribution
- Add OSM basemap

Motivation is here to generate an output HTML map ready-to-publish:

<img src="https://user-images.githubusercontent.com/20856381/213526998-9c710d00-a3a9-48df-af79-7cc26d25d2fe.png" width="700"/>

![image](https://user-images.githubusercontent.com/20856381/213528169-7396ebc7-0556-447b-b478-55a0dadd1970.png)

pathmapper committed:
Add some options for Leaflet html output

nyalldawson commented on python/plugins/processing/algs/qgis/TilesXYZ.py:
```diff
@@ -565,6 +578,21 @@ def initAlgorithm(self, config=None):
                                                                 self.tr('HTML files (*.html)'),
                                                                 optional=True))
+        html_title_param = QgsProcessingParameterString(self.HTML_TITLE,
+                                                        self.tr('Title (Leaflet html output)'),
```
```suggestion
                                                        self.tr('Title (Leaflet HTML output)'),
```

although I'd lean toward something along the lines of "Leaflet HTML output title" / "Leaflet HTML output attribution" / "Include OpenStreetMap basemap in Leaflet HTML output"

pathmapper replied:
Thanks, this sounds much better indeed.

adressed per https://github.com/qgis/QGIS/pull/51522/commits/257fafa006557595e581c43c88a0b9b0dbe8c691

pathmapper committed:
Improve parameter names

Adress code review comment:
https://github.com/qgis/QGIS/pull/51522#discussion_r1083693553
